### ISSUE TITLE
[msgrpc] console.read does not return complete output

### ISSUE THREAD

**Issue by cstayyab** [rapid7/metasploit-framework#12895]
## Steps to reproduce
When calling `console.read` with any bash command such as `cat filename.txt` it does not returns the output of that command.

## Expected behavior
{b'data': b'[*] exec: cat filename.xml\n\n(and the content of filename.txt)', b'prompt': b'\x01\x02msf5\x01\x02 \x01\x02> ', b'busy': False}

## Current behavior
It does not return the output of the bash command. It only returns the output of msfconsole and not the child bash process that is opened to execute that command. It returns the following:

```python
{b'data': b'[*] exec: cat filename.xml\n\n', b'prompt': b'\x01\x02msf5\x01\x02 \x01\x02> ', b'busy': False}
```

### Metasploit version

Framework: 5.0.61-dev-
Console  : 5.0.61-dev-

### I installed Metasploit with:
- [ ] Kali package via apt
- [x] Omnibus installer (nightly)
- [ ] Commercial/Community installer (from http://www.rapid7.com/products/metasploit/download.jsp)
- [ ] Source install (please specify ruby version)

### OS
Distributor ID:	Ubuntu
Description:	Pop!_OS 19.04
Release:	19.04
Codename:	disco

## Related Issue on other Repository
https://github.com/13o-bbr-bbq/machine_learning_security/issues/42#issuecomment-561949169

**Comment by cstayyab** [comment]
Is there any workaround available until this bug is fixed?

**Comment by timwr** [comment]
@cstayyab  perhaps the output is buffered. What happens if you call console.read again?

**Comment by cstayyab** [comment]
@timwr I also thought that and called it again but it also returns empty output.

**Comment by timwr** [comment]
@cstayyab I think this is actually because the child command output ends up being send to stdout/stderr rather than the console buffer. Potentially this is fixable (by redirecting the commands stdout/stderr to the console) but it's not high priority.

**Comment by hhhparty** [comment]
I met with this issue too when I used msf6 and msgrpc. I found some facts:
1. When calling console.write with any bash command such as cat filename.txt, it would export correct result on the msfconsole , if I load msgrpc by msfconsole command. 
2.  But calling console.read with the same console_id, I got the unwanted test , such as "{b'data': b'[*] exec: cat /opt/metasploit-framework/nmap_result_10.10.10.5.xml\n\n', b'prompt': b'msf6 > ', b'busy': False}”

Any help? Thx.
